R/tar_flood_exposure.R

```diff
@@ -100,7 +100,93 @@ zonal_pop_exposure <- function(floodscan_path=fp_fs,
 
 }
 
-
+zonal_pop_exposure_ssn <- function(floodscan_path=fp_fs,
+                               worldpop_inputs =fp_wp,
+                               flood_frac_thresh=0.005,
+                               binarize_floodscan = T,
+                               adm = lgdf_adm$adm1 %>% filter(adm0_pcode == "ET"),
+                               cols_keep = c("adm1_en", "adm1_pcode","adm0_en")
+){
+  tnc_fs <- tidync(floodscan_path)
+  tnc_fs_filt <- fs_filter_bounds(fs_obj = tnc_fs,geometry = adm)
+  r_fs <- fs_to_raster(fs_obj = tnc_fs_filt,band = "SFED_AREA")
+  r_wp <-  mosaic_worldpop(worldpop_inputs,from_path=F)
+  fs_lookup <-  floodscan_lookup(r_fs) 
+  
+  # do the same, but this time by AMJ season
+  lr_max <- unique(fs_lookup$fs_seas) %>%
+    map(
+      \(ssn_tmp){
+        rname_temp<- fs_lookup %>%
+          filter(fs_seas==ssn_tmp) %>%
+          pull(fs_name)
+        r_fs_tmp <- r_fs[[names(r_fs) %in% rname_temp]]
+        r_max <- max(r_fs_tmp)
+        r_max %>%
+          set.names(rname_temp[1])
+        return(r_max)
+      }
+    )
+  r_max <- rast(lr_max)
+  
+  r_fs_max_crop <- crop(r_max, r_wp)
+  ext(r_fs_max_crop) <- ext(r_wp)
+  
+  cat("resampling floodscan to worldpop\n")
+  r_fs_resampled <- terra::resample(x = r_fs_max_crop,r_wp)
+  
+  if(binarize_floodscan){
+    lgl_thresh_mask <- ifel(r_fs_resampled>=flood_frac_thresh,1,0) 
+    r_fs_resampled_masked <- mask(r_fs_resampled,lgl_thresh_mask)
+    r_exposure <- r_fs_resampled_masked * r_wp
```

Comment:
this is not actually binarizing floodscan like i did in SOM HRP methodology. We should check if thats actually what we want to do here. This masking step is not actually doing anything.

To binarize we would want to just run: 

```
r_expsure = lgl_thresh_mask * r_wp
```